src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -461,6 +468,38 @@ public bool ProcessTransaction(Transaction transaction, int? blockHeight = null,
             }
         }
 
+        private bool CleanTransactionsPastMaxReorg(int height)
+        {
+            bool walletUpdated = false;
+
+            if (this.network.Consensus.MaxReorgLength == 0 || this.Wallet.MultiSigAddress.Transactions.Count <= MinimumRetainedTransactions)
+                return walletUpdated;
+
+            int finalisedHeight = height - (int)this.network.Consensus.MaxReorgLength;
+            var pastMaxReorg = new List<TransactionData>();
+            foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions)
```

Comment:
It will be nice to create a `Confirmed` extension here which pre filters the Transactions based on not null spending details and blockheight? But perhaps for another PR ;)